app/services/hearing_schedule/generate_hearing_days_schedule.rb

```diff
@@ -191,59 +191,142 @@ def allocate_all_ro_monthly_hearing_days(ro_key)
+      # results are stored in @ros[ro_key][:allocated_dates]
+      while i < 31 && monthly_allocations.values.inject(:+) != 0
+        i += 1
+        allocate_hearing_days_to_individual_ro(
+          ro_key,
+          monthly_allocations,
+          date_index
+        )
+        date_index += 1
+      end
+    end
+  end
+
+  # Method to assign hearing days for each roomless hearing day requested
+  def assign_hearing_days_without_rooms_to_individual_ro(ro_key, monthly_allocations)
+    # Loop the available alocations for this RO
+    monthly_allocations.each do |month, allocated_days|
+      # Get the available days for this RO this month
+      available_days = @ros[ro_key][:allocated_dates][month]
+
+      # Skip if there are no requested days or available days
+      next if available_days.count == 0 || allocated_days == 0
+
+      # Determine the difference between the requested and available to use as an offset when requested is greater
+      remaining = allocated_days % available_days.count
+
+      # Determine the divisor to use with the offset calculation
+      offset_divisor = (allocated_days < available_days.count) ? allocated_days : remaining
+
+      # Determine whether there is an offset by dividing available by the above calculation
+      offset = (remaining == 0) ? 0 : available_days.count / offset_divisor
+
+      # Initialize the calculated index
+      offset_index = 0
+
+      # Set the nmber allocated to 0 initially
+      @number_to_allocate = 1
+
+      # Loop through the requested number to distribute the hearing days evenly
+      allocated_days.times do
+        # Check whether we have allocated the number to allocate for each day
+        if available_days.values.count { |x| x.count == @number_to_allocate } == available_days.count
+          # Update the number to allocate
+          @number_to_allocate += 1
+        end
+
+        # Determine the index of the date on which we should assign this hearing day
+        offset_index = get_index_for_hearing_day(available_days, offset, offset_index)
+
+        # Add a new hearing day at the index that was calcualted above
+        available_days[available_days.keys[offset_index]].push(room_num: nil)
+      end
+    end
+  end
+
+  # Gets the hearing day index ensuring an even spread of hearing days
+  def get_index_for_hearing_day(available_days, distribution_offset, index)
+    # Calculate the offset based on the distribution offset and the current index
+    offset = distribution_offset + index
+
+    # Calculate the new index with the offset accounting for when the offset exceeds the available days
+    get_index_offset(available_days, offset)
+  end
+
+  # Method to get the index offset for an array ensuring we go around the array instead of getting out of bounds
+  def get_index_offset(available_days, offset)
+    offset_index = (offset >= available_days.count) ? offset - available_days.count : offset
```

Comment:
```suggestion
    offset_index = offset % available_days.length
```
This also works for values of offset >= 2 * available_days.length